### DISCUSSION TITLE
Introduction of PHP CS Fixer has broken PHP 7.4 support

### DISCUSSION THREAD

**Discussion by mikestreety** [deployphp/deployer#3976] [Bugs] [answered]
### Deployer Version

v7.5.6

### Target OS

Alpine

### Which PHP version are you using?

PHP 7.4

### Content of deploy.php or deploy.yaml

N/A

### Steps to reproduce

1. Use PHP 7.4
2. Install >= 7.5.6

Running `dep` in a PHP 7.4 environment throws the following error:

```
Parse error: syntax error, unexpected ')', expecting variable (T_VARIABLE) in phar:///var/www/html/vendor/deployer/deployer/bin/dep/vendor/ringcentral/psr7/src/functions.php on line 355
```

This is because the `php-cs-fixer` appears to have linted the code to PHP 8+, including the files included in the `vendor` folder

https://github.com/deployphp/deployer/commit/44b3a55bd6599985e579e359c759a90a48771f2b

Highlighted in the IDE when inspecting the PHAR contents in the file `vendor/ringcentral/psr7/src/functions.php`.

![image](https://github.com/user-attachments/assets/c5ea9e5d-aea0-49f6-a2a5-cef0556f0945)
 
Although it breaks on this, I suspect other changes in this linted commit will also cause to break

**Comment by mikestreety** [comment]
I can't quite replicate it locally to verify this works (as there seem to be loads of differences in the `vendor` folder when I install the dev dependencies), 

However I believe adding the following rule to `.php-cs-fixer.dist.php` will solve the issue

```
'trailing_comma_in_multiline' => [
            'after_heredoc' => true, 
            'elements' => ['arguments', 'array_destructuring', 'arrays']
]
```

The rules inherit from CSR2.0 which have `'match', 'parameters'` included in the elements, which is for PHP >= 8.0

https://cs.symfony.com/doc/rules/control_structure/trailing_comma_in_multiline.html

I understand this works on `master` as that has a minimum PHP version of 8.2, however version 7 of deployer [still supports PHP 7.3+](https://github.com/deployphp/deployer/blob/7.x/composer.json#L35).

**Comment by Kilbourne** [comment]
Will this be resolved in the next version?

**Comment by Mabahe** [comment]
This discussion is a dupe of still unfixed discussion https://github.com/deployphp/deployer/discussions/3974

**Comment by Kilbourne** [comment]
Added #4001 to fix

**Comment by mikestreety** [comment, marked as answer]
This is now resolved as of [7.5.11](https://github.com/deployphp/deployer/releases/tag/v7.5.11).

Many thanks to @Kilbourne for getting a fix in and @antonmedv for all their hard work with deployer and getting a fix live.
